cli/initiator/resigning.go

```diff
@@ -0,0 +1,130 @@
+package initiator
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	e2m_core "github.com/bloxapp/eth2-key-manager/core"
+	cli_utils "github.com/bloxapp/ssv-dkg/cli/utils"
+	"github.com/bloxapp/ssv-dkg/pkgs/crypto"
+	"github.com/bloxapp/ssv-dkg/pkgs/initiator"
+	"github.com/bloxapp/ssv-dkg/pkgs/wire"
+	"github.com/sourcegraph/conc/pool"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+func init() {
+	cli_utils.SetResigningFlags(StartResigning)
+}
+
+var StartResigning = &cobra.Command{
+	Use:   "resign",
+	Short: "Resigning DKG results",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fmt.Println(`
+		██████╗ ██╗  ██╗ ██████╗     ██████╗ ███████╗███████╗██╗ ██████╗ ███╗   ██╗
+		██╔══██╗██║ ██╔╝██╔════╝     ██╔══██╗██╔════╝██╔════╝██║██╔════╝ ████╗  ██║
+		██║  ██║█████╔╝ ██║  ███╗    ██████╔╝█████╗  ███████╗██║██║  ███╗██╔██╗ ██║
+		██║  ██║██╔═██╗ ██║   ██║    ██╔══██╗██╔══╝  ╚════██║██║██║   ██║██║╚██╗██║
+		██████╔╝██║  ██╗╚██████╔╝    ██║  ██║███████╗███████║██║╚██████╔╝██║ ╚████║
+		╚═════╝ ╚═╝  ╚═╝ ╚═════╝     ╚═╝  ╚═╝╚══════╝╚══════╝╚═╝ ╚═════╝ ╚═╝  ╚═══╝
+																				`)
+
+		if err := cli_utils.SetViperConfig(cmd); err != nil {
+			return err
+		}
+		if err := cli_utils.BindResigningFlags(cmd); err != nil {
+			return err
+		}
+		logger, err := cli_utils.SetGlobalLogger(cmd, "dkg-initiator")
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := cli_utils.Sync(logger); err != nil {
+				log.Printf("Failed to sync logger: %v", err)
+			}
+		}()
+		logger.Info("🪛 Initiator`s", zap.String("Version", cmd.Version))
+		// Load operators
+		opMap, err := cli_utils.LoadOperators(logger)
+		if err != nil {
+			logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
+		}
+		operatorIDs, err := cli_utils.StingSliceToUintArray(cli_utils.OperatorIDs)
+		if err != nil {
+			logger.Fatal("😥 Failed to load participants: ", zap.Error(err))
+		}
+		ethNetwork := e2m_core.NetworkFromString(cli_utils.Network)
+		arrayOfSignedProofs, err := wire.LoadProofs(cli_utils.ProofsFilePath)
+		if err != nil {
+			logger.Fatal("😥 Failed to read proofs json file:", zap.Error(err))
+		}
+		// start the ceremony
+		ctx := context.Background()
+		pool := pool.NewWithResults[*Result]().WithContext(ctx).WithFirstError().WithMaxGoroutines(maxConcurrency)
+		for i := 0; i < len(arrayOfSignedProofs); i++ {
+			i := i
+			pool.Go(func(ctx context.Context) (*Result, error) {
+				// Create new DKG initiator
+				dkgInitiator, err := initiator.New(opMap.Clone(), logger, cmd.Version, cli_utils.ClientCACertPath)
+				if err != nil {
+					return nil, err
+				}
+				// Create a new ID
+				id := crypto.NewID()
+				nonce := cli_utils.Nonce + uint64(i)
+				// Perform the resigning ceremony
+				depositData, keyShares, proofs, err := dkgInitiator.StartResigning(id, operatorIDs, arrayOfSignedProofs[i], ethNetwork, cli_utils.WithdrawAddress.Bytes(), cli_utils.OwnerAddress, nonce)
```

Comment:
remind me if i'm wrong: wasn't the point of proofs to be able to verify the RSA signatures of operators as a confirmation that they approved that file to be used for resiging/resharing? @MatusKysel @pavelkrolevets

the spec has a `ValidateSignedProofs` func, maybe we can use it here right after loading the proofs?

Reply:
we have this check when we write result files to disk: https://github.com/ssvlabs/ssv-dkg/blob/re-sharing/pkgs/validator/validator.go#L135

Reply:
yes, but this is a different check on the new proofs right?

i think we should also validate the proofs before resigning and resharing, so that we wont start a process that we know is gonna be invalid because the proofs are, and we can let the user know of this issue early

Reply:
Ok, agree, added initial proofs validation at initiator before sending messages